apps/email/js/cards/compose.js

```diff
@@ -361,6 +361,30 @@ return [
       };
     },
 
+    checkAddressesPassed: function() {
+      var frobAddressNode = (function(node) {
+        var bubbles = node.parentNode.querySelectorAll('.cmp-peep-bubble');
+        var addrList = [];
+        for (var i = 0; i < bubbles.length; i++) {
+          var dataSet = bubbles[i].dataset;
+          addrList.push(dataSet.address);
+        }
+        if (node.value.trim().length !== 0) {
+          var mailbox = model.api.parseMailbox(node.value);
+          addrList.push(mailbox.address);
+        }
+        return addrList;
+      }.bind(this));
+
+      var to = frobAddressNode(this.toNode).toSource();
+      var cc = frobAddressNode(this.ccNode).toSource();
+      var bcc = frobAddressNode(this.bccNode).toSource();
+
+      return to == this.composer.to.toSource() &&
+        cc == this.composer.cc.toSource() &&
+        bcc == this.composer.bcc.toSource();
+    },
+
```

Comment:
This function could be named more precisely, as the current name does not indicate what test the addresses are passing. Perhaps something like haveAddressesChanged.

It is also tempting to try to just reuse extractAddresses() to do the address extraction, then do some array comparisons to figure out if they changed.

I would prefer to see an array iteration checking the values instead of using toSource(), since toSource is non-standard, and for native arrays could lead to misleading values.

I think it is enough for the check to be something that uses Array.some() to check the values and if there is a difference in the values, return true, and to confirm that the array sizes are the same. That array checking might be a helper function on the prototype since we need to run it for a few arrays.

If you found problems using extractAddresses() for this type of comparison it would be good to know, or if the array comparison suggestion needs more details, may have been a bit too brief in its description.